src/main/java/com/google/api/codegen/discovery/MainDiscoveryProviderFactory.java

```
@@ -129,10 +120,36 @@ public static DiscoveryProvider defaultCreate(
               new RubySnippetSetRunner<Method>(SnippetSetRunner.SNIPPET_RESOURCE_ROOT))
           .setSnippetFileName(id + "/" + DEFAULT_SNIPPET_FILE)
           .build();
+    }
 
-    } else {
+    // Below is the MVVM pathway.
+    SampleMethodToViewTransformer sampleMethodToViewTransformer = null;
+    TypeNameGenerator typeNameGenerator = null;
+    try {
+      sampleMethodToViewTransformer = SAMPLE_METHOD_TO_VIEW_TRANSFORMER_MAP.get(id).newInstance();
+      typeNameGenerator = TYPE_NAME_GENERATOR_MAP.get(id).newInstance();
+    } catch (Exception e) {
+    }
```

why are you hiding the exception?